contracts/staking/StakeManager.sol

```diff
@@ -42,13 +45,17 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
   uint256 public currentEpoch = 1;
   uint256 public NFTCounter = 1;
 
+  enum Status { Inactive, Active, Locked }
+
   struct Validator {
     uint256 epoch;
     uint256 amount;
     uint256 reward;
     uint256 activationEpoch;
     uint256 deactivationEpoch;
     address signer;
+    address contractAddress;
+    Status status;
   }
 
   struct State {
```

Comment:
Lets make the members `uint256`?

Reply:
no, this is the state of timeline so it needs to be signed.